src/main/java/io/searchhub/mph/MPHStringSet.java

```diff
@@ -0,0 +1,164 @@
+package io.searchhub.mph;
+
+import static io.searchhub.mph.MPHUtil.buildEvaluator;
+import static io.searchhub.mph.MPHUtil.getMphFunctionData;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.function.Function;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.minperf.RecSplitEvaluator;
+
+/**
+ * Unmodifiable set with constant size usage, that can only provide the 'contains' and 'size' methods, since the keys are not stored.
+ * It can be used similar to bloom filter: build it in offsite batch process and transported to a low-memory application for filtering.
+ */
+public class MPHStringSet implements Set<String> {
```

Comment:
As we can't call add/remove etc. how about naming this `UnmodifiableMPHStringSet`?